addons/udes_security/models/models.py

```diff
@@ -7,9 +7,27 @@
 class BaseModel(models.AbstractModel):
     _inherit = "base"
 
+    @api.multi
+    def write(self, vals):
+        """Inherit write to add a check against res.archiving.restrictions
+        when the active field is written to.
+        """
+        ResArchivingRestrictions = self.env["res.archiving.restrictions"]
+        if "active" in vals:
```

Comment:
not sure if we also need to check that we are writing the opposite value to any of the records in `self` in case there are some wonky writes going on in certain models which update `"active" : True` when it is already set to True.
If there are and the user triggering those wonky writes doesn't have archiving permission, it will blow up.

Reply:
i was thinking something like this
```python
if "active" in vals:
    current_active = self.mapped("active")
    setting_active = vals.get("active")
    if len(current_active) > 1 or setting_active not in current_active:
        ResArchivingRestrictions._check_can_archive(self._name)
```

Reply:
Will need to try out this scenario, I'm sure in some cases I've seen Odoo filtering out writes that are writing the existing value, but might be bad memory or wishful thinking on my part...

Reply:
I don't think it does unfortunately. But it doesn't matter, let's keep it simple, if the user cannot change the value shouldn't be in vals, no need to only check restriction if the value is different.